AccessCheckoutSDK/AccessCheckoutSDKTests/validation/api/cardBin/CardBinServiceTests.swift

```diff
@@ -371,4 +387,213 @@ class CardBinServiceTests: XCTestCase {
 
         XCTAssertFalse(calledCompletionHandler)
     }
+
+    func testShouldHandleCancellationExceptionGracefully() {
+        let expectation = self.expectation(
+            description: "should handle cancellation exception gracefully")
+        expectation.isInverted = true
+
+        stub(mockClient) { stub in
+            when(stub.abort()).thenDoNothing()
+
+            when(
+                stub.retrieveBinInfo(
+                    cardNumber: any(),
+                    checkoutId: any(),
+                    completionHandler: any()
+                )
+            ).then { _, _, completion in
+
+            }
+        }
+
+        cardBinService.getCardBrands(
+            globalBrand: TestFixtures.discoverBrand(),
+            cardNumber: discoverDinersTestPan
+        ) { _ in
+            expectation.fulfill()
+        }
+
+        Thread.sleep(forTimeInterval: 0.2)
+
+        // will pass if callback was not invoked
+        waitForExpectations(timeout: 1) { _ in
+            verify(self.mockClient).abort()
+            verify(self.mockClient).retrieveBinInfo(
+                cardNumber: any(),
+                checkoutId: any(),
+                completionHandler: any()
+            )
+        }
+    }
+
+    func testShouldNotErrorWhenCallingAbortIfNoCurrentRequestExists() {
```

Comment:
The test name is confusing, it seems to cover the testing of an abort function on the `CardBinService` whereas it covers the `getCardBrands()` function and the `abort()` function of the client. Can we rename it accordingly?